### ISSUE TITLE
[Bug] `relax.frontend.torch.from_exported_program` aborts on sparse CSR buffer (`layout_impl is only implemented for TensorImpl subclasses`)

### ISSUE THREAD

**Issue by tinywisdom** [apache/tvm#18648]
### Summary

When using the TVM Relax Torch frontend to import a `torch.export.export`ed program that involves a `torch.sparse_csr_tensor` registered as a buffer, `from_exported_program` crashes with a C++ `c10::Error`:

> `layout_impl is only implemented for TensorImpl subclasses.`

`torch.export.export` itself succeeds. The crash happens during `from_exported_program(ep)`.

---

### Environment

```text
OS: Linux x86_64
PyTorch: 2.9.0+cu128
TVM: 0.22.0
Python: 3.10.x
```

---

### Minimal Reproduction

```python
# repro_92022_export_tvm_frontend.py
import torch
import torch.nn as nn

import tvm
from tvm.relax.frontend.torch import from_exported_program

print("torch version:", torch.__version__)
print("tvm version:", getattr(tvm, "__version__", "unknown"))


class MyModel(nn.Module):
    def __init__(self):
        super().__init__()
        crow_indices = torch.tensor([0, 1, 2], dtype=torch.int64)
        col_indices = torch.tensor([0, 1], dtype=torch.int64)
        values = torch.tensor([1.0, 1.0], dtype=torch.float32, requires_grad=True)

        csr_tensor = torch.sparse_csr_tensor(
            crow_indices, col_indices, values, dtype=torch.float32
        )
        # Register sparse CSR tensor as a buffer
        self.register_buffer("csr_tensor", csr_tensor)
        # Explicitly enable grad as well
        self.csr_tensor.requires_grad_(True)

    def forward(self, x):
        # Convert buffer to sparse CSR layout again
        csr2 = self.csr_tensor.to_sparse(layout=torch.sparse_csr)
        y = torch.matmul(csr2, x)
        return y.sum()


def GetInput():
    return torch.ones((2, 1), dtype=torch.float32)


def main():
    model = MyModel().to("cpu").eval()
    x = GetInput().to("cpu")

    print("Start torch.export.export ...")
    ep = torch.export.export(model, (x,))
    print("torch.export.export done.")

    print("Start from_exported_program ...")
    ir_mod = from_exported_program(ep)
    print("from_exported_program done.")
    print(ir_mod)


if __name__ == "__main__":
    main()
```

---

### Actual Behavior

Output on my machine:

```text
torch version: 2.9.0+cu128
tvm version: 0.22.0
.../SparseCsrTensorImpl.cpp:53: Sparse CSR tensor support is in beta state...
Start torch.export.export ...
torch.export.export done.
Start from_exported_program ...
terminate called after throwing an instance of 'c10::Error'
  what():  layout_impl is only implemented for TensorImpl subclasses.
Exception raised from layout_impl at /pytorch/c10/core/TensorImpl.h:1094 (most recent call first):
frame #0: c10::Error::Error(...)
frame #1: c10::detail::torchCheckFail(...)
frame #2: ...
frame #3: torch::autograd::InputMetadata::InputMetadata(at::Tensor const&) + ...
frame #4: ...
<libtorch / libtorch_python frames omitted>
...
Aborted (core dumped)
```

`torch.export.export` completes successfully; the abort occurs only when calling `from_exported_program(ep)`.

### Triage

Please refer to the list of label tags [here](https://github.com/apache/tvm/wiki/Issue-Triage-Labels) to find the relevant tags and add them below in a bullet format (example below).

* needs-triage
* bug

**Comment by LudovicoYIN** [comment]
I reproduced the crash locally and confirmed it happens inside exported_program.run_decompositions() when a sparse CSR buffer is present.

Changes (proposed fix):

- In exported_program_translator.py, detect whether the exported program contains sparse tensors (buffers/params/constants/tensor_constants). If so, skip run_decompositions() to avoid the PyTorch layout_impl crash.
- Add a fallback for aten.to_sparse.default in the convert map: treat it as a no‑op, since Relax doesn’t support sparse tensors yet.
- Add a regression test in test_frontend_from_exported_program.py that exports a model with a CSR buffer and ensures from_exported_program completes.
- The minimal repro no longer aborts; import succeeds and the IR is produced.

If this direction looks good, I’m happy to contribute the PR.
